scripts/adminrouter-agent-setup.ps1

```diff
@@ -24,33 +25,32 @@ function New-AdminRouterEnvironment {
         if($LASTEXITCODE) {
             Throw "Failed to delete exiting $ADMINROUTER_SERVICE_NAME service"
         }
-        Write-Output "Deleted existing $ADMINROUTER_SERVICE_NAME service"
+        Write-Log "Deleted existing $ADMINROUTER_SERVICE_NAME service"
     }
     New-Directory -RemoveExisting $ADMINROUTER_DIR
     New-Directory $ADMINROUTER_LOG_DIR
 }
 
-function Get-ApacheServerPackage {
-    Write-Output "Downloading Apache HTTP Server zip file: $WINDOWS_APACHEL_HTTP_SERVER_URL"
-    $filesPath = Join-Path $env:TEMP "httpd-2.4.33-win64-VC15.zip"
-    Start-ExecuteWithRetry { Invoke-WebRequest -UseBasicParsing -Uri $WINDOWS_APACHEL_HTTP_SERVER_URL -OutFile $filesPath}
+function Expand-ApacheServerPackage {
+    $filesPath = Join-Path $AGENT_BLOB_DEST_DIR "httpd-2.4.33-win64-VC15.zip"
+    Write-Log "Expanding Apache HTTP Server zip file $filesPath"
 
     # To be sure that a download is intact and has not been tampered with from the original download side
     # we need to perform biniary hash validation
-    Write-Output "Validating Hash value for downloaded file: $filesPath"
+    Write-Log "Validating Hash value for downloaded file: $filesPath"
     $hashFromDowloadedFile = Get-FileHash $filesPath -Algorithm SHA256
 
     # validate both hashes are the same
     if ($hashFromDowloadedFile.Hash -eq $WINDOWS_APACHEL_HTTP_SERVER_SHA256) {
-        Write-Output 'Hash validation test passed'
+        Write-Log 'Hash validation test passed'
     } else {
         Throw 'Hash validation test FAILED!!'
     }
 
-    Write-Output "Extracting binaries archive in: $ADMINROUTER_DIR"
-    Expand-Archive -LiteralPath $filesPath -DestinationPath $ADMINROUTER_DIR -Force
+    Write-Log "Extracting binaries archive in: $ADMINROUTER_DIR"
+    Expand-7ZIPFile -File $filesPath -DestinationPath $ADMINROUTER_DIR
```

Comment:
We need to add `Expand-7ZIPFile` function to `Utils.psm1` as well. This script execution works because it it called from `DCOSWindowsAgentSetup.ps1` and the function is declared there.

Reply:
I will add it.